util/llbutil/authprovider/cloudauth/authprovider.go

```diff
@@ -287,27 +340,38 @@ func (ap *authProvider) getAuthConfig(ctx context.Context, host string) (*authCo
 
 	// check user's secrets
 	ac, err := ap.getAuthConfigForProject(ctx, "", "", host)
-	if err == nil {
+	if err != nil {
+		if !errors.Is(err, ErrNoCredentialsFound) {
+			return nil, fmt.Errorf("failed to lookup credentials for %s: %w", host, err)
+		}
+	} else {
 		ap.authConfigCache[host] = ac
 		return ac, nil
 	}
```

Comment:
I think it's a little unclear that the `if` block flows back into the main logic, here - the `return` at the end of the `else` and in the nested `if !errors.Is` branch kind of hides that the `if err != nil` case actually merges back to the parent scope. This makes it clearer (to me, at least) how the logic flows into the main logic:

```suggestion
	if err == nil {
		ap.authConfigCache[host] = ac
		return ac, nil
	}
	if !errors.Is(err, ErrNoCredentialsFound) {
		return nil, fmt.Errorf("failed to lookup credentials for %s: %w", host, err)
	}
```